job/server/src/main/java/alluxio/master/job/JobMaster.java

```diff
@@ -388,6 +409,9 @@ public long registerWorker(WorkerNetAddress workerNetAddress) {
       worker.updateLastUpdatedTimeMs();
     }
 
+    jobWorkerHealth.setHostname(hostname);
```

Comment:
Would it be possible to just include/set the hostname when we construct the health object instead of setting it over here?

I feel like it is more bug prone to have the object mostly immutable but then have a single field be able to be set like this. If there is good reason it's fine to keep it

Reply:
When the worker registers The master just uses the information provided in `WorkerNetAddress` - doesn't it report the hostname in there? Shouldn't the worker should have that information as well since it's the source of that information?